### PR TITLE
SAAS-3534: Added Firehose Encrypted Spec File

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/firehose/firehoseEncrypted.spec.js [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        //     firehoseEncrypted.run(cache, {}, (err, results) => {
+        //         console.log(results);
+        //         expect(results.length).to.equal(1);
+        //         expect(results[0].status).to.equal(1);
+        //         expect(results[0].message).to.include('The Firehose delivery stream uses the default KMS key');
+        //         expect(results[0].region).to.equal('us-east-1');
+        //         done();
+        //     });
+        // });
+
```
Comment (sadeed12345) [inline plugins/aws/firehose/firehoseEncrypted.spec.js:268]:
```suggestion
 
```

File: plugins/aws/firehose/firehoseEncrypted.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ module.exports = {
                 var deliveryStreamDesc = describeDeliveryStream.data.DeliveryStreamDescription;
                 var deliveryStreamARN = deliveryStreamDesc.DeliveryStreamARN;
 
-                //console.log(describeDeliveryStream.data.DeliveryStreamDescription.Destinations[0].ExtendedS3DestinationDescription.EncryptionConfiguration.KMSEncryptionConfig);
+                // console.log(JSON.stringify(describeDeliveryStream, null, 2));
```
Comment (AkhtarAmir) [inline plugins/aws/firehose/firehoseEncrypted.js:63]:
```suggestion
```

File: plugins/aws/firehose/firehoseEncrypted.spec.js [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('No Firehose delivery streams found');
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if Unable to query for firehose delivery streams', function (done) {
+            const cache = createCache(null, null, { message: "Unable to Unable to query Firehose for delivery streams" });
```
Comment (dipsubha06) [inline plugins/aws/firehose/firehoseEncrypted.spec.js:270]: Fix the error message

File: plugins/aws/firehose/firehoseEncrypted.spec.js [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+            firehoseEncrypted.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if unable to query Firehose for delivery streams', function (done) {
+            const cache = createCache([listDeliveryStreams[0]], null, { message: "Unable to Unable to query Firehose for delivery streams" });
```
Comment (dipsubha06) [inline plugins/aws/firehose/firehoseEncrypted.spec.js:280]: what is this test case for, as the above one also having same set of params to createcache? is this for describestreamerr, if so then fix the params order in the createcache call.

**Review by dipsubha06 — CHANGES_REQUESTED** [review]

File: plugins/aws/firehose/firehoseEncrypted.spec.js [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+            firehoseEncrypted.run(cache, {}, (err, results) => {
```
Comment (AkhtarAmir) [inline plugins/aws/firehose/firehoseEncrypted.spec.js:281]:
```suggestion
        it('should UNKNOWN if unable to query Firehose for delivery streams', function (done) {
            const cache = createCache([listDeliveryStreams[0]], null, null, { message: "Unable to query Firehose for delivery streams" });
            firehoseEncrypted.run(cache, {}, (err, results) => {
```

File: plugins/aws/firehose/firehoseEncrypted.spec.js (hunk 4 above)
Comment (AkhtarAmir) [inline plugins/aws/firehose/firehoseEncrypted.spec.js:280]:
```suggestion
            const cache = createCache([listDeliveryStreams[0]], null, { message: "Unable to query for firehose delivery streams" });
```

**Pull request merged**